src/chrome/CommitBar.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ReactFragment } from 'react'
+
+import Spinner from "./Spinner"
+import TextInput from "./forms/TextInput"
+import Button from "./Button"
+import Checkbox from "./forms/Checkbox"
+
+interface CommitBarProps {
+  // used for the message displayed on the left side of the bar
+  commitBarContent: JSX.Element | ReactFragment
+  // shows a spinner in the commit button
+  commitLoading: boolean
+  // value for the commit title input
+  commitTitle: string
+  // fired when the title input is edited
+  onCommitTitleChange: (d: string) => void
+  // fired when the commit button is clicked
+  onCommit: () => void
+  // whether to show the Run now checkbox
+  showRunNow?: boolean
+  // the value of the Run now checkbox
+  runNow?: boolean
+  // fired when the Run now checkbox is changed
+  onRunNowChange?: () => void
+}
+
+const CommitBar: React.FC<CommitBarProps> = ({
```

Comment:
We should reuse this component in `DatasetEditorLayout`

Reply:
Great catch!